src/ovi/Provider.php

```diff
@@ -0,0 +1,468 @@
+
+           $username = runtime_sanatizeItem::sanatizeItem(trim($params->username),"string");
+           $username_valid = $this->ovipanel_UsernameExists($params->username);
+
+           if($username_valid ==  'false')
+           {
+        	$error = 'Account revokeReseller Failed. Username '.$username.' does not exists.';
+	        return $this->errorResult('Error: ' . $error);
+           }
+
+            $input_tag='<username>'.$username.'</username>';
+            $return = $this->ovipanel_CurlExecute( $input_tag, "revokeReseller");
+
+            $data = json_decode($return);
+                if ($data === null) 
+		{
+		        $error = "Failed to decode JSON";
+		        return $this->errorResult('Error: ' . $error);
+            	}
+	         elseif ($data->success) {
+		   return ResellerPrivileges::create()->setMessage('Reseller privileges revoked')->setReseller(false);
+          	}
+    }
+
+    /**
+     * @inheritDoc
+     */
+
+      public function sanatizeItem($var, $type)
+        {
+                if($var == '' || isset($var) == NULL)
+                return NULL ;
+                $flags = NULL;
+                switch($type)
+                {
+                        case 'url':
+                        $filter = FILTER_SANITIZE_URL;
+                        break;
+
+                        case 'int':
+                        $filter = FILTER_SANITIZE_NUMBER_INT;
+                        break;
+
+                        case 'pass':
+                        $filter = FILTER_UNSAFE_RAW;
+                        $flags = null;
+                        break ;
+
+                        case 'IP':
+                        $filter = FILTER_VALIDATE_IP;
+                        $flags = FILTER_FLAG_IPV4;
+                        break;
+
+                        case 'email':
+                        $var = substr($var, 0, 254);
+                        $filter = FILTER_SANITIZE_EMAIL;
+                        break;
+
+                        case 'string':
+                        $filter = FILTER_SANITIZE_STRING;
+                        break;
+
+                        case 'boolean':
+                        $filter = FILTER_VALIDATE_BOOLEAN;
+                        break;
+
+                        default:
+                        $filter = FILTER_SANITIZE_STRING;
+                        $flags = FILTER_FLAG_NO_ENCODE_QUOTES;
+                        break;
+                }
+
+        $output = filter_var($var, $filter, $flags);
+        if($type != 'pass'){
+                $output = str_replace(array('\\', "\0", "\n", "\r", "'", '"', "\x1a","=","`"), array('', '', '', '', '', '', '','','','',''), $output);
+        }else{
+                $output = str_replace(array('\\', "\0", "\n", "\r", "\x1a","`"), array('', '', '', '', '', '', '','','','',''), $output);
+        }
+                return($output);
+        }
+
+
+    public function ovipanel_CurlExecute($input_tag, $request)
```

Comment:
Please use either use PHPDoc blocks or strict parameter and return types in all methods to allow for static analysis (e.g., in IDEs)